File: cocos2D/Lib/Ads/AdColony/iOS/SampleApps/ObjC/AdColonyV4VC/AdColonyV4VC/ViewController.m
//
//  ViewController.m
//  AdColonyV4VC
//
//  Created by Owain Moss on 1/14/15.
//  Copyright (c) 2015 AdColony. All rights reserved.
//

#import "ViewController.h"

#import "Constants.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIImageView *background;
@property (weak, nonatomic) IBOutlet UIButton *button;
@property (weak, nonatomic) IBOutlet UILabel  *currencyLabel;
@property (weak, nonatomic) IBOutlet UILabel  *statusLabel;
@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *spinner;
@end


@implementation ViewController

#pragma mark - UIViewController overrides

- (void)viewDidLoad {
    [super viewDidLoad];
    
    if (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad) {
        [_spinner setActivityIndicatorViewStyle:UIActivityIndicatorViewStyleWhiteLarge];
    } else {
        [_spinner setActivityIndicatorViewStyle:UIActivityIndicatorViewStyleWhite];
    }
    
    [self updateCurrencyBalance];
    [self zoneLoading];
    
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(removeObservers) name:UIApplicationDidEnterBackgroundNotification object:nil];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(addObservers) name:UIApplicationWillEnterForegroundNotification object:nil];
    [self addObservers];
}

- (UIStatusBarStyle)preferredStatusBarStyle {
    return UIStatusBarStyleLightContent;
}

- (UIInterfaceOrientationMask)supportedInterfaceOrientations {
    return UIInterfaceOrientationMaskAll;
}

- (BOOL)shouldAutorotate {
    return YES;
}

#pragma mark - UI Updates

- (void)addObservers {
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(updateCurrencyBalance) name:kCurrencyBalanceChange object:nil];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(zoneReady) name:kZoneReady object:nil];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(zoneOff) name:kZoneOff object:nil];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(zoneLoading) name:kZoneLoading object:nil];
}

- (void)removeObservers {
    [[NSNotificationCenter defaultCenter] removeObserver:self name:kZoneLoading object:nil];
    [[NSNotificationCenter defaultCenter] removeObserver:self name:kZoneOff object:nil];
    [[NSNotificationCenter defaultCenter] removeObserver:self name:kZoneReady object:nil];
    [[NSNotificationCenter defaultCenter] removeObserver:self name:kCurrencyBalanceChange object:nil];
}

- (void)zoneLoading {
    [_spinner setHidden:NO];
    [_spinner startAnimating];
    [_button  setAlpha:0.];
    [UIView animateWithDuration:.5 animations:^{
        _statusLabel.alpha = 1.;
    } completion:nil];
}

- (void)zoneReady {
    [_spinner stopAnimating];
    [_spinner setHidden:YES];
    [_statusLabel setAlpha:0.];
    [UIView animateWithDuration:1.0 animations:^{
        _button.alpha = 1.;
    } completion:nil];
}

- (void)zoneOff {
    [_spinner stopAnimating];
    [_button  setAlpha:0.];
    [_statusLabel setText:@"V4VC disabled!"];
}

- (void)updateCurrencyBalance {
    //Get currency balance from persistent storage and display it
    NSNumber* wrappedBalance = [[NSUserDefaults standardUserDefaults] objectForKey:kCurrencyBalance];
    NSUInteger balance = wrappedBalance && [wrappedBalance isKindOfClass:[NSNumber class]] ? [wrappedBalance unsignedIntValue] : 0;
    [_currencyLabel setText:[NSString stringWithFormat:@"%lu", (unsigned long)balance]];
}

#pragma mark - AdColony

- (IBAction)triggerVideo {
    [AdColony playVideoAdForZone:kAdColonyZoneID withDelegate:self withV4VCPrePopup:YES andV4VCPostPopup:YES];
}

- (void)onAdColonyAdAttemptFinished:(BOOL)shown inZone:(NSString *)zoneID {
    if (!shown && [AdColony zoneStatusForZone:kAdColonyZoneID] != ADCOLONY_ZONE_STATUS_ACTIVE) {
        [self zoneLoading];
    } else if (!shown) {
        [self zoneReady];
    }
}
@end

File: cocos2D/Lib/Ads/AdColony/iOS/SampleApps/Swift/AdColonyV4VC/AdColonyV4VC/ViewController.swift
//
//  ViewController.swift
//  AdColonyV4VC
//
//  Created by Owain Moss on 1/15/15.
//  Copyright (c) 2015 AdColony. All rights reserved.
//

import UIKit

class ViewController: UIViewController, AdColonyAdDelegate
{
    @IBOutlet weak var background:    UIImageView!
    @IBOutlet weak var button:        UIButton!
    @IBOutlet weak var text:          UIImageView!
    @IBOutlet weak var currencyLabel: UILabel!
    @IBOutlet weak var statusLabel:   UILabel!
    @IBOutlet weak var spinner:       UIActivityIndicatorView!
    
    //=============================================
    // MARK:- UIViewController Overrides
    //=============================================
    
    override func viewDidLoad()
    {
        super.viewDidLoad()
        
        if UIDevice.currentDevice().userInterfaceIdiom == .Pad {
            self.spinner.activityIndicatorViewStyle = UIActivityIndicatorViewStyle.WhiteLarge
        } else {
            self.spinner.activityIndicatorViewStyle = UIActivityIndicatorViewStyle.White
        }
        
        self.updateCurrencyBalance()
        
        NSNotificationCenter.defaultCenter().addObserver(self, selector: "addObservers", name: UIApplicationWillEnterForegroundNotification, object: nil)
        NSNotificationCenter.defaultCenter().addObserver(self, selector: "removeObservers", name: UIApplicationDidEnterBackgroundNotification, object: nil)
        self.addObservers()
    }
    
    override func preferredStatusBarStyle() -> UIStatusBarStyle
    {
        return UIStatusBarStyle.LightContent
    }
    
    override func supportedInterfaceOrientations() -> UIInterfaceOrientationMask
    {
        return UIInterfaceOrientationMask.All
    }
    
    override func shouldAutorotate() -> Bool
    {
        return true
    }
    
    //=============================================
    // MARK:- UI Updates
    //=============================================
    
    func addObservers()
    {
        NSNotificationCenter.defaultCenter().addObserver(self, selector: "updateCurrencyBalance", name: Constants.currencyBalanceChange, object: nil)
        NSNotificationCenter.defaultCenter().addObserver(self, selector: "zoneReady", name: Constants.zoneReady, object: nil)
        NSNotificationCenter.defaultCenter().addObserver(self, selector: "zoneOff", name: Constants.zoneOff, object: nil)
        NSNotificationCenter.defaultCenter().addObserver(self, selector: "zoneLoading", name: Constants.zoneLoading, object: nil)
    }
    
    func removeObservers()
    {
        NSNotificationCenter.defaultCenter().removeObserver(self, name: Constants.zoneLoading, object: nil)
        NSNotificationCenter.defaultCenter().removeObserver(self, name: Constants.zoneOff, object: nil)
        NSNotificationCenter.defaultCenter().removeObserver(self, name: Constants.zoneReady, object: nil)
        NSNotificationCenter.defaultCenter().removeObserver(self, name: Constants.currencyBalanceChange, object: nil)
    }
    
    func zoneReady()
    {
        self.spinner.stopAnimating()
        self.spinner.hidden = true
        self.statusLabel.alpha = 0;
        UIView.animateWithDuration(1.0, animations: {
            self.button.alpha  = 1
        })
    }
    
    func zoneOff()
    {
        self.spinner.stopAnimating()
        self.button.alpha = 0;
        self.statusLabel.text = "V4VC disabled!"
    }
    
    func zoneLoading()
    {
        self.spinner.hidden = false
        self.spinner.startAnimating()
        self.button.alpha = 0
        UIView.animateWithDuration(0.5, animations: {
            self.statusLabel.alpha = 1.0;
        })
    }
    
    func updateCurrencyBalance()
    {
        //Get currency balance from persistent storage and display it
        if let wrappedBalance = NSUserDefaults.standardUserDefaults().objectForKey(Constants.currencyBalance) as! NSNumber? {
            self.currencyLabel.text = wrappedBalance.stringValue
        } else {
            self.currencyLabel.text = "0"
        }
    }
   
    //=============================================
    // MARK:- AdColony
    //=============================================    
    
    @IBAction func triggerVideo(sender: AnyObject)
    {
        AdColony.playVideoAdForZone(Constants.adcolonyZoneID, withDelegate: self, withV4VCPrePopup: true, andV4VCPostPopup: true)
    }
    
    func onAdColonyAdAttemptFinished(shown: Bool, inZone zoneID: String)
    {
        if !shown && AdColony.zoneStatusForZone(Constants.adcolonyZoneID) != ADCOLONY_ZONE_STATUS.ACTIVE {
            self.zoneLoading()
        } else if !shown {
            self.zoneReady()
        }
    }
}
